### ISSUE TITLE
PoX2 stacking Bug list

### ISSUE THREAD

**Issue by timstackblock** [hirosystems/stacks-wallet#1149]
@markmhx I will log all the bugs from testing into 1 issue instead of making multiple issues.

1) Wallet shows 0 stacking but I have over 5 million STX stacked
<img width="1119" alt="Screen Shot 2023-01-09 at 11 36 36 AM" src="https://user-images.githubusercontent.com/49165468/211359819-1158cf16-12eb-4ef2-a65c-95f95dcb55d9.png">

**Comment by markmhx** [comment]
Thanks for the report! Is this on the Stacks 2.1 testnet or regular testnet?

**Comment by timstackblock** [comment]
That screenshot is from the regular testnet still waiting on STX for the 2.1 testnet

<img width="530" alt="Screen Shot 2023-01-09 at 12 20 33 PM" src="https://user-images.githubusercontent.com/49165468/211368637-46596609-05ef-4c2f-a722-2e6a1a0fd5ad.png">

Also would I be able to interact with any of the new functionality in this build as well.

<img width="804" alt="Screen Shot 2023-01-09 at 12 22 49 PM" src="https://user-images.githubusercontent.com/49165468/211369018-55de1d3b-7872-4dc0-b72e-420bca03a2cf.png">

**Comment by edu-stx** [comment]
> Also would I be able to interact with any of the new functionality in this build as well.

We're not planning on providing this functionality for the desktop wallet. It's feature set will remain the same as 2.1 goes live. We'll make the new functionality available through a [new web app currently being built](https://github.com/hirosystems/stacking).

**Comment by timstackblock** [comment]
Got it ok thanks.

 The number stacked is also showing on testnet 2.1 so that works as well.

<img width="1120" alt="Screen Shot 2023-01-09 at 1 10 54 PM" src="https://user-images.githubusercontent.com/49165468/211378207-2ed38b5b-3d7f-4c46-a7de-d245af3cfe7f.png">

**Comment by timstackblock** [comment]
> We're not planning on providing this functionality for the desktop wallet. Its feature set will remain the same as 2.1 goes live. We'll make the new functionality available through a [new web app currently being built](https://github.com/hirosystems/stacking).


The new web app is not ready yet correct? Thats still in development, do we have a ballpark figure of when that will be done?

**Comment by timstackblock** [comment]
Hey guys quick question I stacked yesterday with this transaction for 1 cycle https://2-1-api.testnet.hiro.so/extended/v1/tx/0xb7b20646f55d864498ed5944b355cf463a244a42e556d9b1793e755a0ee967ff
The cycle is for 1 day but my wallet’s stacking progress  is only 12.4% complete. I would usually look in the explorer to see if the progress matched but since it is not loading, I am trying to figure out why the stacking  is not further along if cycles are only 1 day on the 2.1 testnet. 

<img width="639" alt="Screen Shot 2023-01-10 at 1 54 59 PM" src="https://user-images.githubusercontent.com/49165468/211638077-2d6c8807-9f81-4c55-8610-65627a5ecf03.png">

<img width="445" alt="Screen Shot 2023-01-10 at 1 55 04 PM" src="https://user-images.githubusercontent.com/49165468/211638109-6b8edc1e-f371-4bee-a21b-b3c9001d27dd.png">

**Comment by timstackblock** [comment]
Also what API endpoint are you using to get data for "Progress"

**Comment by edu-stx** [comment]
@timstackblock

> The new web app is not ready yet correct? Thats still in development, do we have a ballpark figure of when that will be done?

That's correct. Unfortunately I don't have a very precise estimate for its completion, most likely a few weeks out. Its being built as both an updated and stripped down port of the current desktop wallet app, with support for the wallet extension, and I'm slowly learning as the port progresses

> API endpoint [...] for "Progress"

Seems that it is being calculated from the data returned by the account balance, which can be retrieved with [`.getAccountBalances`](https://github.com/hirosystems/stacks.js/tree/master/packages/stacking#get-account-balances-from-api). The calculation is performed here,

https://github.com/hirosystems/stacks-wallet/blob/83bb80e6bd660877770bbe475665fa667ee3a125/app/store/stacking/stacking.reducer.ts#L244

**Comment by timstackblock** [comment]
Well I think that is correct, I got more stacks and was able to get back into the stacking flow the issue is the wallet just incorrectly states the number of days in the cycle see images below. This is where the confusion came from.

FOR STACKING IT STATES ITS 1 DAY

<img width="613" alt="Screen Shot 2023-01-11 at 2 41 31 PM" src="https://user-images.githubusercontent.com/49165468/211902444-295ff78f-d7ad-4257-8b66-0e72b6404aba.png">

FOR DELEGATION IT STATES THE CYCLES START OVER EVERYDAY

<img width="583" alt="Screen Shot 2023-01-11 at 2 38 08 PM" src="https://user-images.githubusercontent.com/49165468/211902609-c3d09b5d-41d7-4d6a-810e-0fa3cfd14490.png">
<img width="583" alt="Screen Shot 2023-01-11 at 2 38 13 PM" src="https://user-images.githubusercontent.com/49165468/211902611-1d3842d1-da2f-4b45-8894-b1273572063c.png">
<img width="589" alt="Screen Shot 2023-01-11 at 2 38 19 PM" src="https://user-images.githubusercontent.com/49165468/211902615-611ca548-9933-4262-a5ea-02376bd6fda5.png">

**Comment by markmhx** [comment]
@timstackblock we tried to recreate the bug reported initially at the top of this issue with v4.8.0 (https://github.com/hirosystems/stacks-wallet/issues/1149#issue-1525935125) but weren't able to do so. Is it still showing 0 to the right for you?

**Comment by edu-stx** [comment]
@timstackblock seems the value of 0 stx locked is coming from the API. For example, with a test account I just stacked, yet the API is reporting 0,

Request URL,

```
Request URL: https://stacks-node-api.testnet.stacks.co/v2/accounts/ST1Q38DQ4EAXEP1YDSAV4C0PBPZF34TTK5MAKCQQN?proof=0
```

Response data,

```json
{
    "balance":"0x0000000000000000000005a749fb9dc0",
    "locked":"0x00000000000000000000000000000000",
    "unlock_height":0,
    "nonce":34
}
```

However, other endpoints seem to correctly report the right amounts,

Request,

```
https://stacks-node-api.testnet.stacks.co/extended/v1/address/ST1Q38DQ4EAXEP1YDSAV4C0PBPZF34TTK5MAKCQQN/balances
```

Response data,

```json
{
  "stx": {
    "balance": "6216058800000",
    "total_sent": "1341000000",
    "total_received": "6217402000000",
    "total_fees_sent": "2200000",
    "total_miner_rewards_received": "0",
    "lock_tx_id": "0x60ca37bf8f02239dd7cbdad6aa22ae5a54ae0c8cb62b8357ad107f047b8430a8",
    "locked": "6000000000000",
    "lock_height": 93666,
    "burnchain_lock_height": 2416152,
    "burnchain_unlock_height": 2417900
  },
  "fungible_tokens": {},
  "non_fungible_tokens": {
    "ST000000000000000000002AMW42H.bns::names": {
      "count": "1",
      "total_sent": "0",
      "total_received": "1"
    }
  }
}
```

**Comment by edu-stx** [comment]
Opened https://github.com/hirosystems/stacks-blockchain-api/issues/1520 to track API issue.

**Comment by markmhx** [comment]
@edu-stx is this days-calculation issue client or API-side? https://github.com/hirosystems/stacks-wallet/issues/1149#issuecomment-1379395684

**Comment by timstackblock** [comment]
It looks like 0 STX locked isn't an issue, it seems to be waiting for the cycle to start when it started it updated.

**Comment by timstackblock** [comment]
The only thing that is an issue is the cycle being reported as 1 day

**Comment by markmhx** [comment]
@timstackblock Do we need to do any particular testing for when the same network activates PoX2? To make sure the wallet gracefully switches over from PoX1 to PoX2? Or is that akin to switching between the PoX1 testnet to PoX2 testnet manually?

**Comment by timstackblock** [comment]
I would delegate a small amount of STX to a pool to make sure the PoX2 contract is correctly configured in the wallet.

**Comment by timstackblock** [comment]
If you want we can jump on a call and talk through some of the scenarios your concerned about and I can loop in someone from the API or Blockchain team.
